packages/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashScreen.ts

```diff
@@ -29,7 +34,18 @@ export const withAndroidSplashScreen: ConfigPlugin = config => {
   return withPlugins(config, [
     withAndroidSplashImages,
     [withAndroidSplashDrawables, splashConfig],
-    withAndroidSplashMainActivity,
+    ...(shouldUpdateLegacyMainActivity(config) ? [withAndroidSplashLegacyMainActivity] : []),
     withAndroidSplashStyles,
+    withAndroidSplashStrings,
   ]);
 };
+
+function shouldUpdateLegacyMainActivity(config: ExpoConfig): boolean {
+  try {
+    const projectRoot = config._internal?.projectRoot;
+    const packagePath = resolveFrom(projectRoot, 'expo-splash-screen/package.json');
+    const version = JsonFile.read(packagePath).version?.toString() ?? '';
+    return semver.lt(version, '0.12.0');
+  } catch {}
+  return false;
```

Comment:
```suggestion
  return config.sdkVersion && semver.lt(config.sdkVersion, '43.0.0');
```

If the user doesn't have expo-splash-screen installed, they'll still the defaults to be based on the template they have. We version the templates based on SDK version so we can check that -- if the SDK Version isn't defined, or it's invalid, we should default to the latest functionality (so maybe add a try/catch).